### PR TITLE
fix(zh-tw): replace the references to broken image source

### LINKED ISSUE #17951: zh-TW: All pictures cannot be displayed normally.
### MDN URL

https://developer.mozilla.org/zh-TW/docs/Web/CSS/CSS_flexible_box_layout/Basic_concepts_of_flexbox

### What specific section or headline is this issue about?

_No response_

### What information was incorrect, unhelpful, or incomplete?

All pictures cannot be displayed normally.
截圖 2024-01-18 下午3 52 43
截圖 2024-01-18 下午3 52 57

### What did you expect to see?

All pictures can be displayed normally.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `zh-tw/web/css/css_flexible_box_layout/basic_concepts_of_flexbox`
* MDN URL: https://developer.mozilla.org/zh-TW/docs/Web/CSS/CSS_flexible_box_layout/Basic_concepts_of_flexbox
* GitHub URL: https://github.com/mdn/translated-content/blob/main/files/zh-tw/web/css/css_flexible_box_layout/basic_concepts_of_flexbox/index.md
* Last commit: https://github.com/mdn/translated-content/commit/d49a658f5fa3fa3e5cb3332ccde9cc06215e074a
* Document last modified: 2023-08-02T16:33:15.000Z

</details>

### PR DESCRIPTION
**Pull request by yin1999**
### Description

replace the references to broken image source.

The imported image is from [Internet Archive](https://web.archive.org/web/20210309193826/https://developer.mozilla.org/zh-TW/docs/Web/CSS/CSS_flexible_box_layout/Basic_concepts_of_flexbox).

### Related issues and pull requests

Part of: #19088
Fixes: #17951

### CODE REVIEW

**Review by yin1999 — COMMENTED** [review]
Note: the current PR does not synchronize the translation, only repair/remove the content.

File: files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md [hunk 1]
```diff
@@ -245,98 +245,106 @@ init();
...
-    clearX = CanvasXSize;
-  }
-  if (imgH > CanvasYSize) {
-    clearY = imgH;
-  } // image larger than canvas
-  else {
-    clearY = CanvasYSize;
+
+  if (imgW > canvasXSize) {
+    // Image larger than canvas
```
Comment (JasonLamv-t) [inline files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md:290]:
```suggestion
    // 圖像大於畫布
```

File: files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md [hunk 2]
```diff
@@ -245,98 +245,106 @@ init();
...
+    x = canvasXSize - imgW;
   }
-  //Get Canvas Element
+
+  // Check if image dimension is larger than canvas
```
Comment (JasonLamv-t) [inline files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md:294]:
```suggestion
  // 檢查圖像尺寸是否大於畫布
```

File: files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md [hunk 3]
```diff
@@ -245,98 +245,106 @@ init();
...
+  clearX = Math.max(imgW, canvasXSize);
+  clearY = Math.max(imgH, canvasYSize);
+
+  // Get canvas context
```
Comment (JasonLamv-t) [inline files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md:298]:
```suggestion
  // 取得畫布上下文
```

File: files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md [hunk 4]
```diff
@@ -245,98 +245,106 @@ init();
...
   ctx = document.getElementById("canvas").getContext("2d");
-  //Set Refresh Rate
+
+  // Set refresh rate
```
Comment (JasonLamv-t) [inline files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md:301]:
```suggestion
  // 設定刷新率
```

File: files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md [hunk 5]
```diff
@@ -245,98 +245,106 @@ init();
...
-本例會產一個由左到右循環捲動[美國優勝美地國家公園](http://commons.wikimedia.org/wiki/File:Capitan_Meadows,_Yosemite_National_Park.jpg)景色，你也可以自行替換其他比畫布還大的圖片。
+本例會產一個由左到右循環捲動[美國優勝美地國家公園](https://commons.wikimedia.org/wiki/File:Capitan_Meadows,_Yosemite_National_Park.jpg)景色，你也可以自行替換其他比畫布還大的圖片。
+
+### HTML
+
+循環景色就是在下方的 {{HTMLElement("canvas")}} 中捲動，請注意其中的 width 和 height 和程式碼中的 `canvasXSize` 與 `canvasYSize` 一樣。
+
+```html
+<canvas id="canvas" width="800" height="200"
+  >Yosemite National Park, meadow at the base of El Capitan</canvas
```
Comment (JasonLamv-t) [inline files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md:258]:
```suggestion
  >優勝美地國家公園，埃爾卡皮坦山腳下的草地</canvas
```

File: files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md [hunk 6]
```diff
@@ -245,98 +245,106 @@ init();
...
-  //Clear Canvas
-  ctx.clearRect(0, 0, clearX, clearY);
-  //If image is <= Canvas Size
-  if (imgW <= CanvasXSize) {
-    //reset, start from beginning
-    if (x > CanvasXSize) {
-      x = 0;
+  ctx.clearRect(0, 0, clearX, clearY); // clear the canvas
+
+  // If image is <= canvas size
```
Comment (JasonLamv-t) [inline files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md:308]:
```suggestion
  // 如果圖像 <= 畫布大小
```

File: files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md [hunk 7]
```diff
@@ -245,98 +245,106 @@ init();
...
+  if (imgW <= canvasXSize) {
+    // Reset, start from beginning
```
Comment (JasonLamv-t) [inline files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md:310]:
```suggestion
    // 重置，從頭開始
```

File: files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md [hunk 8]
```diff
@@ -245,98 +245,106 @@ init();
...
+    if (x > canvasXSize) {
+      x = -imgW + x;
     }
-    //draw aditional image
-    if (x > CanvasXSize - imgW) {
-      ctx.drawImage(img, x - CanvasXSize + 1, y, imgW, imgH);
+
+    // Draw additional image1
```
Comment (JasonLamv-t) [inline files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md:315]:
```suggestion
    // 繪製附加圖片1
```

File: files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md [hunk 9]
```diff
@@ -245,98 +245,106 @@ init();
...
+      ctx.drawImage(img, -imgW + x, y, imgW, imgH);
     }
-  }
-  //If image is > Canvas Size
-  else {
-    //reset, start from beginning
-    if (x > CanvasXSize) {
-      x = CanvasXSize - imgW;
+
+    // Draw additional image2
```
Comment (JasonLamv-t) [inline files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md:320]:
```suggestion
    // 繪製附加圖片2
```

File: files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md [hunk 10]
```diff
@@ -245,98 +245,106 @@ init();
...
+    if (x - imgW > 0) {
+      ctx.drawImage(img, -imgW * 2 + x, y, imgW, imgH);
+    }
+  } else {
+    // Image is > canvas size
+    // Reset, start from beginning
```
Comment (JasonLamv-t) [inline files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md:326]:
```suggestion
    // 圖片 > 畫布尺寸
    // 重置，從頭開始
```

File: files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md [hunk 11]
```diff
@@ -245,98 +245,106 @@ init();
...
+    if (x > canvasXSize) {
+      x = canvasXSize - imgW;
     }
-    //draw aditional image
-    if (x > CanvasXSize - imgW) {
+
+    // Draw additional image
```
Comment (JasonLamv-t) [inline files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md:331]:
```suggestion
    // 繪製附加圖片
```

File: files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md [hunk 12]
```diff
@@ -245,98 +245,106 @@ init();
...
+    if (x > canvasXSize - imgW) {
       ctx.drawImage(img, x - imgW + 1, y, imgW, imgH);
     }
   }
-  //draw image
+
+  // Draw image
```
Comment (JasonLamv-t) [inline files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md:337]:
```suggestion
  // 繪製圖片
```

File: files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md [hunk 13]
```diff
@@ -245,98 +245,106 @@ init();
...
   ctx.drawImage(img, x, y, imgW, imgH);
-  //amount to move
+
+  // Amount to move
```
Comment (JasonLamv-t) [inline files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md:340]:
```suggestion
  // 移動量
```

**Review by JasonLamv-t — COMMENTED** [review]
Some suggestions.

File: files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md [hunk 14]
```diff
@@ -245,98 +245,106 @@ init();
...
-var scale = 1.05;
-var y = -4.5; //vertical offset
+img.src = "capitan_meadows_yosemite_national_park.jpg";
+const canvasXSize = 800;
+const canvasYSize = 200;
+const speed = 30; // lower is faster
+const scale = 1.05;
+const y = -4.5; // vertical offset
 
 // Main program
```
Comment (yin1999) [inline files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md:276]:
```suggestion
// 主程式
```

File: files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md [hunk 15]
```diff
@@ -245,98 +245,106 @@ init();
...
-var CanvasYSize = 200;
-var speed = 30; //lower is faster
-var scale = 1.05;
-var y = -4.5; //vertical offset
+img.src = "capitan_meadows_yosemite_national_park.jpg";
+const canvasXSize = 800;
+const canvasYSize = 200;
+const speed = 30; // lower is faster
+const scale = 1.05;
+const y = -4.5; // vertical offset
```
Comment (yin1999) [inline files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md:274]:
```suggestion
const speed = 30; // 越小越快
const scale = 1.05;
const y = -4.5; // 垂直偏移
```

File: files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md [hunk 16]
```diff
@@ -245,98 +245,106 @@ init();
...
+```
+
+### JavaScript
 
 ```js
-var img = new Image();
+const img = new Image();
 
 // User Variables - customize these to change the image being scrolled, its
 // direction, and the speed.
```
Comment (yin1999) [inline files/zh-tw/web/api/canvas_api/tutorial/basic_animations/index.md:268]:
```suggestion
// 使用者變數——自訂這些變數以更改滾動的圖像、方向和速度。
```

**Review by JasonLamv-t — APPROVED** [review]
LGTM. Thank you! @yin1999

**Pull request merged**